rotkehlchen/db/dbhandler.py

```diff
@@ -2813,6 +2813,12 @@ def query_timed_balances(
         cursor = self.conn.cursor()
         results = cursor.execute(querystr, bindings)
         results = results.fetchall()
+
+        if settings.eth_equivalent_eth2 and asset.identifier == 'ETH':
+            bindings = [from_ts, to_ts, 'ETH2']
+            eth2_results = cursor.execute(querystr, bindings)
+            results.extend(eth2_results.fetchall())
```

Comment:
This is again completely wrong.

1. You are doing a special query only for ETH and not for ETH2
2. You are doing 2 database queries.
3. You are extending the previous result, destroying the established timestamp order that is shown in the graph, thus breaking the entire thing.

What you need to do is conditionally append to the previous query and make it query for currency = ETH and currency = ETH2 if the setting is on and ETH is given as the `asset` argument and then take the sum of the amount of both, grouping by timestamp ....

example:

```SELECT time, SUM(amount), usd_value, category FROM timed_balances WHERE time BETWEEN 1654369489 AND 1654808691 AND (currency="ETH" OR currency="ETH2") GROUP BY time;```

Also if ETH2 is given and the setting is on, return nothing.

Reply:
```py
querystr = (
    'SELECT time, amount, usd_value, category FROM timed_balances '
    'WHERE time BETWEEN ? AND ? AND (currency=?'
)
bindings = [from_ts, to_ts, asset.identifier]

if settings.eth_equivalent_eth2 and asset.identifier == 'ETH':
    querystr += ' OR currency=?)'
    bindings.append('ETH2')
else:
    querystr += ')'
``` 
I've done something like this, is this okay?

Reply:
no this is not okay. This will not work. I **literally** wrote to you the exact query that needs to be used. Do you see what you are missing?

Reply:
Ah sorry I've included the sum in the code, I wrote it here so forgot to keep it in this snippet.